stm32-modules/include/flex-stacker/flex-stacker/tmf8821.hpp

```diff
@@ -0,0 +1,457 @@
+#pragma once
+
+#include <cstdint>
+
+#include "firmware/tmf8820_image.h"
+#include "flex-stacker/tmf8821_registers.hpp"
+#include "systemwide.h"
+
+namespace tmf8821 {
+using namespace tof::hardware;
+
+template <typename P>
+concept TMF8821Policy = requires(P p, uint16_t dev_addr, uint16_t reg,
+                                 uint16_t size, uint8_t* data) {
+    { p.i2c_read(dev_addr, reg, size) } -> std::same_as<RxTxReturn>;
+    { p.i2c_write(dev_addr, reg, data, size) } -> std::same_as<RxTxReturn>;
+};
+
+// I2C Device address
+constexpr uint16_t TOF_DEFAULT_ADDRESS = 0x41;
+constexpr uint16_t TOF_X_ADDRESS = 0x39;
+constexpr uint16_t TOF_Z_ADDRESS = 0x40;
+// Frame retry defaults
+constexpr uint8_t DEFAULT_RETRIES = 5;
+constexpr uint32_t DEFAULT_SLEEP_MS = 250;
+
+// Bootloader commands (bl_cmd_stat)
+
+// Remap RAM to Address 0 and Reset
+constexpr uint8_t BL_RAMREMAP_RESET = 0x11;
+// Initialize for RAM download from host to TMF8820/21/28
+constexpr uint8_t BL_DOWNLOAD_INIT = 0x14;
+// Build in self-test of RAM (pattern test)
+constexpr uint8_t BL_RAM_BIST = 0x2A;
+// Build in self-test of I²C RAM (pattern test)
+constexpr uint8_t BL_I2C_BIST = 0x2C;
+// Write RAM Region (Plain = not encoded into e.g. Intel Hex Records)
+constexpr uint8_t BL_W_RAM = 0x41;
+// Set the read/write RAM pointer to a given address
+constexpr uint8_t BL_ADDR_RAM = 0x43;
+
+// Bootloader cmd status
+enum BL_CMD_STATUS {
+    STAT_READY = 0,
+    STAT_ERR_SIZE,
+    STAT_ERR_CSUM,
+    STAT_ERR_RANGE,
+    STAT_ERR_MORE,
+};
+
+// Bootloader Payload constants
+constexpr uint8_t BL_DATA_LEN = 128;  // chunk len
+constexpr uint8_t BL_HEADER_LEN = 2;  // cmd + len
+constexpr uint8_t BL_FOOTER_LEN = 1;  // checksum
+constexpr uint8_t BL_BUFFER_LEN = BL_HEADER_LEN + BL_DATA_LEN + BL_FOOTER_LEN;
+uint8_t BUFFER[BL_BUFFER_LEN] = {0};
```

Comment:
We don't want this in the header file; it's fine right now because I guess this header is only in one translation unit, but were we ever to include it in another translation unit we'd get linker duplicate-symbol errors because we're violating the one-definition rule. We can declare it here but we need to define it in a source file or inside a class def or something.